plugins/modules/azure_rm_publicipprefix_info.py

```diff
@@ -0,0 +1,281 @@
+            returned: always
+            type: list
+            sample: ['1', '2']
+        prefix_length:
+            description:
+                - The Length of the Public IP Prefix.
+            type: int
+            returned: always
+            sample: 29
+        extended_location:
+            description:
+                - The extended location of the public ip address.
+            type: str
+            returned: always
+            sample: 'eastus2'
+        custom_ip_prefix:
+            description:
+                - The customIpPrefix that this prefix is associated with.
+            type: dict
+            returned: always
+            sample: {}
+'''
+try:
+    from azure.core.exceptions import ResourceNotFoundError
+except Exception:
+    # This is handled in azure_rm_common
+    pass
+
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
+
+AZURE_OBJECT_CLASS = 'PublicIpPrefix'
+
+
+class AzureRMPublicIPPrefixInfo(AzureRMModuleBase):
+
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            name=dict(type='str'),
+            resource_group=dict(type='str'),
+            tags=dict(type='list', elements='str')
+        )
+
+        self.results = dict(
+            changed=False,
+        )
+
+        self.name = None
+        self.resource_group = None
+        self.tags = None
+
+        super(AzureRMPublicIPPrefixInfo, self).__init__(self.module_arg_spec,
+                                                        supports_check_mode=True,
+                                                        supports_tags=False,
+                                                        facts_module=True)
+
+    def exec_module(self, **kwargs):
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+
+        result = []
+        if self.name is not None and self.resource_group is not None:
+            result = self.get_item()
+        elif self.resource_group:
+            result = self.list_resource_group()
+        else:
+            result = self.list_all()
+
+        raw = self.filter(result)
+
+        self.results['publicipprefixes'] = self.format(raw)
+
+        return self.results
+
+    def format(self, raw):
+        return [self.prefix_to_dict(item) for item in raw]
+
+    def filter(self, response):
+        return [item for item in response if self.has_tags(item.tags, self.tags)]
+
+    def prefix_to_dict(self, prefix):
+        result = dict(
+            id=prefix.id,
+            name=prefix.name,
+            tags=prefix.tags,
+            type=prefix.type,
+            location=prefix.location,
+            public_ip_address_version=prefix.public_ip_address_version,
+            prefix_length=prefix.prefix_length,
+            provisioning_state=prefix.provisioning_state,
+            etag=prefix.etag,
+            zones=prefix.zones,
+            sku=dict(),
+            ip_tags=dict(),
```

Comment:
```suggestion
            ip_tags=list(),
```

this is populated to be a list in line 247, so it should likely also be an empty list as default.

Reply:
should this be a list of dicts though?
it would probably make more sense to mirror the implementation of `tags` for `ip_tags` and just have a dict?

Reply:
i guess it should probably be kept as list to be aligned with `azure_rm_publicipaddress`, which then just needs the change commented above.

Reply:
Yes, ip_tags's type is 'list', elements is dict!